packages/story-editor/src/utils/contrastUtils.js

```diff
@@ -158,3 +158,26 @@ export function getAccessibleTextColorsFromPixels(pixelData, fontSize) {
     backgroundColor: { ...blackRgb, a: 0.7 },
   };
 }
+
+/**
+ * Gets ImageData object representing the underlying pixel data for a specified portion of the canvas
+ *
+ * @param {Object} pixelData Array of pixel data.
+ * @param ctx
+ * @param {number} x x-axis coordinate of the top-left corner of the area from which the ImageData will be extracted.
+ * @param {number} y y-axis coordinate of the top-left corner of the area from which the ImageData will be extracted.
+ * @param {number} width width of the area from which the ImageData will be extracted.
+ * @param {number} height height of the area from which the ImageData will be extracted.
+ * @param {number} sharpness resolution to use in getting the pixels data.
+ * @return {Object} Returns ImageData object containing the image data for the canvas context and coordinates specified
+ */
+export function getPixelsFromCoords(ctx, x, y, width, height, sharpness) {
```

Comment:
@Trapsta Could you explain what exactly is the change here? I'm not sure I understand how this excludes the selected text element from the contrast calculation.

Currently, when testing, the contrast is doesn't seem right anymore, e.g. adding a new Title in this PR doesn't result in good contrast, part of the title is on a darker background and that doesn't seem to be considered (generally I'm not able to see a highlight being added anymore even when it's needed):
<img width="180" alt="Screenshot 2021-11-16 at 09 35 53" src="https://user-images.githubusercontent.com/3294597/141941163-363d0354-d8a0-4765-a0a3-1e48bfac0da0.png">

Adding a title in `main` with the same image -- highlight is added to the background for ensuring the text to stand out:
<img width="219" alt="Screenshot 2021-11-16 at 09 36 29" src="https://user-images.githubusercontent.com/3294597/141941240-e25701b1-2392-4152-afdf-f65e64cda1f6.png">

The contrast calculation was fine as it was before, there's no need to change that part. The problem was that when re-triggering the logic, the element itself was also included to the image that was used for calculations, influencing the result. As far as I can see that still happens -- the element is still included in the image and it still influences the result.

There shouldn't be a need to change the general contrast calculation, we should find a way how to exclude the selected element from the image that `htmlToImage` is generating for contrast calculation in case of re-triggering the logic specifically.

Please let me know if you have any questions about it / if anything is unclear.

Reply:
Thanks @miina , that provides more clarity but I think the steps to reproduce on the issue description might be a bit misleading. We might need to update the description with the context you provided here.

Reply:
@Trapsta The steps to reproduce usually just show one example of how to reproduce the issue.

The description and the expected behaviour are the things to check in case of any issue. E.g. here the expected behaviour is:
"The element itself should not be included into the generated image."

Reply:
Anytime you have doubts or are thinking of a solution that does not satisfy the expected behaviour / does not fully fix the described issue, please reach out to discuss or leave a comment under the issue.

Reply:
Cool, noted. Thanks